athena-msk/src/main/java/com/amazonaws/athena/connectors/msk/AmazonMskRecordHandler.java

```diff
@@ -176,23 +200,28 @@ private void consume(
      * @param queryStatusChecker - instance of {@link QueryStatusChecker}
      * @param splitParameters - instance of {@link SplitParameters}
      * @param record - instance of {@link ConsumerRecord}
+     * @return boolean - true if all the values are spilled else false
      */
-    private void execute(
+    private boolean execute(
             BlockSpiller spiller,
             ReadRecordsRequest recordsRequest,
             QueryStatusChecker queryStatusChecker,
             SplitParameters splitParameters,
             ConsumerRecord<String, TopicResultSet> record)
     {
+        final boolean[] isExecuted = {false};
```

Comment:
Why is this an array? Only one element is ever written to or read from.

Also when you make this back into a normal boolean, you should probably remove the final modifier there's no reason for it to be final if you want to write to it.

Reply:
Variable used in lambda expression should be final or effectively final - that's why we have used this approach.

Alternate way we can do the following:

        AtomicBoolean isExecuted = new AtomicBoolean(false);
        spiller.writeRows((Block block, int rowNum) -> {
            for (MSKField field : record.value().getFields()) {
                boolean isMatched = block.offerValue(field.getName(), rowNum, field.getValue());
                if (!isMatched) {
                    return 0;
                }
            }
            // For debug insight
            splitParameters.spilled += 1;
            isExecuted.set(true);
            return 1;
        });
        return isExecuted.get();


Let us know your thought.

Reply:
Since all you're using this boolean for is logging, can you just use the logger on line 222 and not use a boolean at all (and also of course just turn it back into `void execute(...)`).

Reply:
Changed accordingly